File: pkg/lumina/components/shadcn/button.lua
-- shadcn/button — Terminal-adapted Button component
local lumina = require("lumina")

local Button = lumina.defineComponent({
    name = "ShadcnButton",
    init = function(props)
        return {
            variant = props.variant or "default",
            size = props.size or "default",
            disabled = props.disabled or false,
            label = props.label or "Button",
        }
    end,
    render = function(self)
        local styles = {
            default      = { bg = "#2563EB", fg = "#FFFFFF" },
            outline      = { bg = "",        fg = "#E2E8F0" },
            secondary    = { bg = "#1E293B", fg = "#E2E8F0" },
            ghost        = { bg = "",        fg = "#E2E8F0" },
            destructive  = { bg = "#DC2626", fg = "#FFFFFF" },
            link         = { bg = "",        fg = "#3B82F6" },
        }
        local s = styles[self.variant] or styles.default

        local sizeMap = {
            default = { paddingLeft = 1, paddingRight = 1, height = 3 },
            sm      = { padding = 0, height = 1 },
            lg      = { padding = 2, height = 7 },
            icon    = { padding = 0, width = 3, height = 3 },
            xs      = { padding = 0, height = 1 },
        }
        local sz = sizeMap[self.size] or sizeMap.default

        return {
            type = "hbox",
            style = {
                background = s.bg,
                foreground = self.disabled and "#666666" or s.fg,
                border = "rounded",
                padding = sz.padding,
                paddingLeft = sz.paddingLeft,
                paddingRight = sz.paddingRight,
                height = sz.height,
                width = sz.width,
                justify = "center",
                align = "center",
                dim = self.disabled,
            },
            children = self.props and self.props.children or {
                { type = "text", content = self.label }
            },
        }
    end
})

return Button

File: pkg/widget/button.go
package widget

// Deprecated: This Go widget is superseded by lua/lux/button.lua which provides
// richer features (severity, appearance, group, split) without Go dependency.
// The Go widget remains for backward compatibility but will be removed in a future version.

import "github.com/akzj/lumina/pkg/render"

// ButtonState is the internal state for a Button widget.
type ButtonState struct {
	Hovered bool
	Pressed bool
}

// Button is the built-in Button widget.
var Button = &Widget{
	Name: "Button",
	NewState: func() any {
		return &ButtonState{}
	},
	Render: func(props map[string]any, state any) any {
		t := CurrentTheme
		s := state.(*ButtonState)
		label, _ := props["label"].(string)
		if label == "" {
			label = "Button"
		}
		disabled, _ := props["disabled"].(bool)
		variant, _ := props["variant"].(string)
		if variant == "" {
			variant = "primary"
		}

		// Determine colors based on variant + state
		bg := t.Primary
		fg := t.PrimaryDark
		switch variant {
		case "outline":
			bg = ""
			fg = t.Primary
		case "ghost":
			bg = ""
			fg = t.Text
		case "secondary":
			bg = t.Surface1
			fg = t.Text
		}

		if disabled {
			bg = t.Surface0
			fg = t.Muted
		} else if s.Pressed {
			bg = t.Pressed
		} else if s.Hovered {
			bg = t.Hover
		}

		// Determine border style
		border := "rounded"
		if variant == "ghost" {
			border = "none"
		}

		textNode := &render.Node{
			Type:    "text",
			Content: label,
			Style: render.Style{
				Foreground: fg,
				Right:      -1,
				Bottom:     -1,
			},
		}
		if disabled {
			textNode.Style.Dim = true
		}

		boxNode := &render.Node{
			Type:      "hbox",
			Children:  []*render.Node{textNode},
			Focusable: !disabled,
			Disabled:  disabled,
			Style: render.Style{
				Border:       border,
				Background:   bg,
				PaddingLeft:  1,
				PaddingRight: 1,
				Justify:      "center",
				Align:        "center",
				Right:        -1,
				Bottom:       -1,
			},
		}
		textNode.Parent = boxNode

		return boxNode
	},
	OnEvent: func(props map[string]any, state any, event *render.WidgetEvent) bool {
		s := state.(*ButtonState)
		disabled, _ := props["disabled"].(bool)
		if disabled {
			return false
		}

		switch event.Type {
		case "mouseenter":
			if !s.Hovered {
				s.Hovered = true
				return true // state changed, needs re-render
			}
		case "mouseleave":
			if s.Hovered || s.Pressed {
				s.Hovered = false
				s.Pressed = false
				return true
			}
		case "mousedown":
			if !s.Pressed {
				s.Pressed = true
				return true
			}
		case "mouseup":
			if s.Pressed {
				s.Pressed = false
				return true
			}
		}
		return false
	},
}
